File: apps/sepuling-java/src/main/java/smecalculus/rolevod/messaging/SepulkaClientImpl.java
package smecalculus.rolevod.messaging;

import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import smecalculus.rolevod.core.SepulkaService;
import smecalculus.rolevod.messaging.SepulkaMessageEm.RegistrationRequest;
import smecalculus.rolevod.messaging.SepulkaMessageEm.RegistrationResponse;
import smecalculus.rolevod.validation.EdgeValidator;

@RequiredArgsConstructor
public class SepulkaClientImpl implements SepulkaClient {

    @NonNull
    private EdgeValidator validator;

    @NonNull
    private SepulkaMessageMapper mapper;

    @NonNull
    private SepulkaService service;

    @Override
    public RegistrationResponse register(RegistrationRequest requestEdge) {
        validator.validate(requestEdge);
        var request = mapper.toDomain(requestEdge);
        var response = service.register(request);
        return mapper.toEdge(response);
    }
}

File: apps/sepuling-kotlin/src/main/kotlin/smecalculus/rolevod/messaging/SepulkaClientImpl.kt
package smecalculus.rolevod.messaging

import smecalculus.rolevod.core.SepulkaService
import smecalculus.rolevod.messaging.SepulkaMessageEm.RegistrationRequest
import smecalculus.rolevod.messaging.SepulkaMessageEm.RegistrationResponse
import smecalculus.rolevod.validation.EdgeValidator

class SepulkaClientImpl(
    private val validator: EdgeValidator,
    private val mapper: SepulkaMessageMapper,
    private val service: SepulkaService,
) : SepulkaClient {
    override fun register(requestEdge: RegistrationRequest): RegistrationResponse {
        validator.validate(requestEdge)
        val request = mapper.toDomain(requestEdge)
        val response = service.register(request)
        return mapper.toEdge(response)
    }
}
